eex/translators/amber/amber_write.py

```diff
@@ -59,6 +59,25 @@ def _write_amber_data(file_handle, data, category):
     _write_1d(file_handle, np.array(data), ncols, fmt)
 
 
+def _get_charmm_dihedral_count(dl):
+    # TODO this a temporary solution to get the number of dihedrals stored in the datalayer.
+    order = 4
+    ret = 0
+    charmm = amd.forcefield_parameters['dihedral']['form']
+    terms = dl.list_term_parameters(order)
+    for j in terms.values():
+        term_md = eex.metadata.get_term_metadata(order, "forms", j[0])
+        # Zip up the parameters
+        parameters = {k: v for k, v in zip(term_md["parameters"], j[1:])}
+        parameters = eex.form_converters.convert_form(order, parameters, j[0], charmm)
```

Comment:
It seems that `convert_form` should always return the same data type. If it is 1-to-1 perhaps it should still return an array with one parameter.